conda/cli/main_env_create.py

```diff
@@ -0,0 +1,209 @@
+    from ..gateways.disk.delete import rm_rf
+    from ..misc import touch_nonadmin
+    from . import install as cli_install
+
+    spec = specs.detect(
+        name=args.name,
+        filename=get_filename(args.file),
+        directory=os.getcwd(),
+        remote_definition=args.remote_definition,
+    )
+    env = spec.environment
+
+    # FIXME conda code currently requires args to have a name or prefix
+    # don't overwrite name if it's given. gh-254
+    if args.prefix is None and args.name is None:
+        args.name = env.name
+
+    prefix = determine_target_prefix(context, args)
+
+    if args.yes and prefix != context.root_prefix and os.path.exists(prefix):
+        rm_rf(prefix)
+    cli_install.check_prefix(prefix, json=args.json)
+
+    # TODO, add capability
+    # common.ensure_override_channels_requires_channel(args)
+    # channel_urls = args.channel or ()
+
+    result = {"conda": None, "pip": None}
+
+    args_packages = (
+        context.create_default_packages if not args.no_default_packages else []
+    )
+
+    if args.dry_run:
+        installer_type = "conda"
+        installer = get_installer(installer_type)
+
+        pkg_specs = env.dependencies.get(installer_type, [])
+        pkg_specs.extend(args_packages)
+
+        solved_env = installer.dry_run(pkg_specs, args, env)
+        if args.json:
+            print(json.dumps(solved_env.to_dict(), indent=2))
+        else:
+            print(solved_env.to_yaml(), end="")
+
+    else:
+        if args_packages:
+            installer_type = "conda"
+            installer = get_installer(installer_type)
+            result[installer_type] = installer.install(prefix, args_packages, args, env)
+
+        if len(env.dependencies.items()) == 0:
+            installer_type = "conda"
+            pkg_specs = []
+            installer = get_installer(installer_type)
+            result[installer_type] = installer.install(prefix, pkg_specs, args, env)
+        else:
+            for installer_type, pkg_specs in env.dependencies.items():
+                try:
+                    installer = get_installer(installer_type)
+                    result[installer_type] = installer.install(
+                        prefix, pkg_specs, args, env
+                    )
+                except InvalidInstaller:
+                    e = (
+                        dals(
+                            f"""
+                        Unable to install package for {0}.
+
+                        Please double check and ensure your dependencies file has
+                        the correct spelling.  You might also try installing the
+                        conda-env-{0} package to see if provides the required
+                        installer.
+                        """
+                        )
+                        .lstrip()
+                        .format(installer_type)
+                    )
+                    sys.stderr.write(e)
+
+                    return -1
```

Comment:
```suggestion
                except InvalidInstaller:
                    raise CondaError(
                        dals(
                            f"""
                        Unable to install package for {installer_type}.

                        Please double check and ensure your dependencies file has
                        the correct spelling. You might also try installing the
                        conda-env-{installer_type} package to see if provides
                        the required installer.
                        """
                        )
                        .lstrip()
                    )
```

Reply:
Thank you! I also did the same thing for `main_env_update`

Reply:
Super!

Reply:
`dals` already includes a `str.lstrip`

Reply:
Oh awesome, TIL!

Will update accordingly. 👍

Reply:
`dals` stands for "dedent and left strip"